### PR TITLE
Upkeep

### LINKED ISSUE #29: Depreicated functions in called R package
Below is a warning thrown when min_depth_interaction functions() is used.

> interactions_frame <- min_depth_interactions(rf_fit, vars)

Warning messages:
1: `funs()` was deprecated in dplyr 0.8.0.
ℹ Please use a list of either functions or lambdas:

# Simple named list: list(mean = mean, median = median)

# Auto named with `tibble::lst()`: tibble::lst(mean, median)

# Using lambdas list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))
ℹ The deprecated feature was likely used in the randomForestExplainer package.
  Please report the issue to the authors.

### LINKED ISSUE #30: Link in README does not work
link to master thesis in README does not work, that is:
[Master thesis on randomForestExplainer](https://cdn.staticaly.com/gh/geneticsMiNIng/BlackBoxOpener/master/randomForestExplainer_Master_thesis.pdf).

### PR DESCRIPTION
**Pull request by olivroy**
Fix #29, fix #30 + other modernization of the package

Replace reshape2 by tidyr 

Closes #32

All the warnings in https://modeloriented.github.io/randomForestExplainer/reference/plot_min_depth_interactions.html are now gone.

The only warnings that remain in tests is:
  Forest grown in ranger version <0.11.5, converting ...

### CODE REVIEW

File: docs/reference/plot_importance_rankings-1.png
Comment (olivroy) [inline docs/reference/plot_importance_rankings-1.png:1]: The graph seems to have changed. Is this expected or should I look if I broke something accidentally?

File: docs/reference/plot_min_depth_interactions-1.png
Comment (olivroy) [inline docs/reference/plot_min_depth_interactions-1.png:1]: Same here

File: docs/reference/plot_importance_ggpairs-1.png
Comment (olivroy) [inline docs/reference/plot_importance_ggpairs-1.png:1]: Same here

**Review by olivroy — COMMENTED** [review]
I will take a look to see if the changes are expected or not.

Edit: I compared outputs/ plots from master branch. did not see changes

The most prominent changes seem to originate from scales/ggplot2 updates, so breaks changed slightly, but no big problems!!

**Comment by pbiecek** [comment]
@olivroy  wow, this is huge! Thank you for your contribution

**Pull request merged**